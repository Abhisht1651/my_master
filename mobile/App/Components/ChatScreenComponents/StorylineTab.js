// /@flow
import React from 'react'
import { View, Text, FlatList } from 'react-native'
import styles from '../Styles/ChatScreenComponentStyles/StorylineTabStyles'
import { Images } from '../../Themes'
import Storyline from '../StoryLineShortesCard'

type Props = {
  item: {
    category: string,
    storyTitle: string,
    date: string,
    description: string,
    source: string,
    trending: boolean,
    storyImage: Image,
    date: string,
    storyline: {
      category: string,
      storyTitle: string,
      date: string,
      description: string,
      source: string,
      trending: boolean,
      storyImage: Image
    },
    id: string
  }
}

const Data = [
  {
    id: '111',
    date: 'Today',
    storyline: [
      {
        id: '1',
        seen: 50,
        total: 500,
        storyTitle: 'UK exit from the European Union',
        date: 'Sept 21, 2019',
        description: 'Ray Contreras talks about the different accents',
        source: 'Forbes',
        trending: true,
        category: 'World',
        storyImage: Images.storyImage
      },
      {
        id: '2',
        seen: 95,
        total: 500,
        storyTitle: 'UK exit from the European Union',
        date: 'Sept 21, 2019',
        description: 'Ray Contreras talks about the different accents',
        source: 'Forbes',
        trending: true,
        category: 'World',
        storyImage: Images.storyImage
      },
      {
        id: '3',
        seen: 155,
        total: 500,
        storyTitle: 'UK exit from the European Union',
        date: 'Sept 21, 2019',
        description: 'Ray Contreras talks about the different accents',
        source: 'Forbes',
        trending: true,
        category: 'World',
        storyImage: Images.storyImage
      }
    ]
  },
  {
    id: '1111',
    date: 'Jan 15th 2020',
    storyline: [
      {
        id: '11',
        seen: 50,
        total: 500,
        storyTitle: 'UK exit from the European Union',
        date: 'Sept 21, 2019',
        description: 'Ray Contreras talks about the different accents',
        source: 'Forbes',
        trending: true,
        category: 'World',
        storyImage: Images.storyImage
      },
      {
        id: '12',
        seen: 95,
        total: 500,
        storyTitle: 'UK exit from the European Union',
        date: 'Sept 21, 2019',
        description: 'Ray Contreras talks about the different accents',
        source: 'Forbes',
        trending: true,
        category: 'World',
        storyImage: Images.storyImage
      }
    ]
  }
]

const StorylineTab = () => {
  const renderstoryline = ({ item }: Props) => {
    return (
      <View style={styles.storylinemargin}>
        <Storyline
          storyTitle={item.storyTitle}
          date={item.date}
          description={item.description}
          source={item.source}
          storyImage={item.storyImage}
          trending={item.trending}
          category={item.category}
        />
      </View>
    )
  }
  const renderitem = ({ item }: Props) => {
    return (
      <View>
        <View style={styles.flatlist1cont}>
          <Text style={styles.date}>{item.date}</Text>
          <FlatList
            data={item.storyline}
            keyExtractor={(post) => post.id}
            renderItem={renderstoryline}
            listKey={(post) => `${post.id}D`}
          />
        </View>
        <View style={styles.border} />
      </View>
    )
  }
  return (
    <View style={styles.container}>
      <View style={styles.margincontainer}>
        <FlatList
          data={Data}
          keyExtractor={(item) => item.id}
          renderItem={renderitem}
          listKey={(item) => `${item.id}D`}
        />
      </View>
    </View>
  )
}

export default StorylineTab
