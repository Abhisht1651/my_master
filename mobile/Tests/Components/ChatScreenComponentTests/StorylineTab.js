import React from 'react'
import { shallow } from 'enzyme'
import { FlatList } from 'react-native'
import StorylineTab from '../../../App/Components/ChatScreenComponents/StorylineTab'

describe('MessageScreen', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<StorylineTab />)
  })
  it('MediaTab component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 flatlist', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
})
