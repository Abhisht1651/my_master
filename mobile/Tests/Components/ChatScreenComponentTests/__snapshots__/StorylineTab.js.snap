// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageScreen MediaTab component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 50,
        "marginRight": 50,
      }
    }
  >
    <FlatList
      data={
        Array [
          Object {
            "date": "Today",
            "id": "111",
            "storyline": Array [
              Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "id": "1",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "id": "2",
                "seen": 95,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "id": "3",
                "seen": 155,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
            ],
          },
          Object {
            "date": "Jan 15th 2020",
            "id": "1111",
            "storyline": Array [
              Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "id": "11",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "id": "12",
                "seen": 95,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      listKey={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</View>
`;
